rotkehlchen/chain/ethereum/modules/yearn/utils.py

```diff
@@ -72,30 +73,62 @@ def _maybe_reset_yearn_cache_timestamp(data: dict[str, Any] | None) -> bool:
     return False  # will continue
 
 
+def _merge_data_yearn_vaults() -> tuple[list[dict[str, Any]] | None, str | None]:
+    """At the moment of writing this logic ydemon doesn't support yearn v1 so we
+    need to aggregate the information from two different apis and remove duplicates.
+    """
+    msg = None
+    data: list[dict[str, Any]] = []
+    timeout_tuple = CachedSettings().get_timeout_tuple()
+    for api_url in (YEARN_OLD_API, YDEMON_API):
+        try:
+            response = requests.get(api_url, timeout=timeout_tuple)
+        except requests.exceptions.RequestException as e:
+            msg = f'Failed to obtain yearn vault information. {e!s}'
+
+        if response.status_code in (HTTPStatus.NOT_FOUND, HTTPStatus.SERVICE_UNAVAILABLE):
+            msg = 'Failed to obtain a response from the yearn API'
+        else:
+            try:
+                new_data = response.json()
+            except (DeserializationError, JSONDecodeError) as e:
+                msg = f"Failed to deserialize data from yearn's old api. {e!s}"
+            else:
+                if not isinstance(new_data, list):
+                    msg = f'Unexpected format from yearn vaults response. Expected a list, got {data}'  # noqa: E501
+                else:
+                    data.extend(response.json())
+
+        if msg is not None:
+            return None, msg
+
+    deduplicated_data, vaults_seen = [], set()
+    for vault in data:
+        if (vault_address := vault.get('address')) is None:
+            log.error(f'Unexpected vault schema in yearn api for {vault}. Skipping')
+            continue
+
+        if (version := vault.get('version')) is not None and version.startswith('3.'):
+            log.debug(f'Skipping yearn v3 vault {vault.get("address")}')
+            continue  # skip v3 vaults until we add them #7540
+
+        if vault_address in vaults_seen:
```

Comment:
why not make them both sets and avoid this extra if? Since you are adding it if it's not there already and that's a set. So that's not needed right?